pkg/config/generate.go

```diff
@@ -327,11 +328,19 @@ func Generate(cs *acsapi.OpenShiftManagedCluster) (err error) {
 		}
 	}
 
-	if len(c.HtPasswd) == 0 {
-		if c.HtPasswd, err = makeHtPasswd("demo", "demo"); err != nil {
-			return
+	// TODO: Remove these password operations before GA
+	if len(c.AdminPasswd) == 0 {
+		if c.AdminPasswd, err = randomString(10); err != nil {
+			return err
 		}
 	}
+	tHtPasswd, err := makeHtPasswd("osadmin", c.AdminPasswd)
+	if err != nil {
+		return err
+	}
+	if bcrypt.CompareHashAndPassword(c.HtPasswd, tHtPasswd) != nil {
```

Comment:
can you confirm you checked this - I expect CompareHashAndPassword to need a hash and a password as arguments, I don't see how it would work with two hashes.

Reply:
sorry, just now got this AHA moment what you had in mind.